src/v3/src/components/AuthenticatorButton/AuthenticatorButton.tsx

```diff
@@ -100,6 +104,79 @@ const AuthenticatorButton: UISchemaElementComponent<{
     return <ArrowRightIcon titleAccess={ctaLabel} />;
   }
 
+  function renderDescription() {
+    if (gracePeriodRequiredDescription && gracePeriodExpiry) {
+      return (
+        <Box
+          sx={{
+            display: 'flex',
+            margin: tokens.Spacing0,
+            marginBlockStart: tokens.Spacing1,
+          }}
+        >
+          <ClockIcon
+            sx={{
+              color: tokens.TypographyColorWarning,
+            }}
+          />
+          <Box
+            sx={{
+              marginLeft: tokens.Spacing1,
+            }}
+          >
+            <Typography
+              paragraph
+              data-se="authenticator-grace-period-required-description"
+              sx={{
+                fontSize: tokens.TypographySizeSubordinate,
+                fontWeight: tokens.TypographyWeightBodyBold,
+                color: tokens.TypographyColorWarning,
+                margin: tokens.Spacing0,
+                textAlign: 'start',
+              }}
+            >
+              {gracePeriodRequiredDescription}
+            </Typography>
+            <Typography
+              paragraph
+              data-se="authenticator-grace-period-expiry-date"
+              sx={{
+                fontSize: tokens.TypographySizeSubordinate,
+                fontWeight: tokens.TypographyWeightBody,
+                color: tokens.TypographyColorSubordinate,
+                margin: tokens.Spacing0,
+                textAlign: 'start',
+              }}
+            >
+              {gracePeriodExpiry}
+            </Typography>
+          </Box>
+        </Box>
+      );
+    }
+    return description
+      && (
+        <Typography
+          paragraph
+          id={`${iconName}-description`}
+          sx={{
+            fontSize: tokens.TypographySizeSubordinate,
+            fontWeight: tokens.TypographyWeightBody,
+            color: tokens.TypographyColorSubordinate,
+            margin: tokens.Spacing0,
+            marginBlockStart: tokens.Spacing1,
+            textAlign: 'start',
+          }}
+          data-se="authenticator-button-description"
+          aria-label={punctuate(description)}
+          dir={dir}
+          translate={noTranslate ? 'no' : undefined}
+        >
+          {description}
+        </Typography>
+      );
+  }
```

Comment:
Not sure if TS complained but you should `return null` at the bottom as the default if nothing should render

Reply:
I was seeing some lint errors that I resolved, but updated to make it return an explicit null